tests/test_password_sign_on.py

```diff
@@ -50,100 +57,133 @@ def crazy_casing(netid: str) -> str:
+        fresh_browser.wait_for_tag('u', 'Miscellaneous')
+        with fresh_browser.autocapture_off():
+            fresh_browser.wait_for_tag('strong', 'Session Expiration (barring inactivity):')
+    fresh_browser.close()
 
 
-# PWD 9, 12, 13
-@pytest.mark.parametrize('login_transform, password_transform, expected_error_message', [
-    # Adds an unsupported suffix to the user's netid
-    (partial(add_suffix, '@google.com'), lambda p: p, 'please sign in with your uw netid'),
-    # Transforms the user's netid to become one that doesn't exist
-    (partial(add_suffix, 'zzz'), lambda p: p, 'your sign-in failed'),
-    # Reverses the user's password, so that it is incorrect
-    (lambda n: n, lambda p: p[::-1], 'your sign-in failed'),
+@pytest.mark.parametrize('login_transform', [
+    partial(add_suffix, '@u.washington.edu'),
+    partial(add_suffix, '@washington.edu'),
+    partial(add_suffix, '@uw.edu'),
+    partial(add_suffix, '@google.com')
 ])
-def test_password_sign_on_fails(
-        utils: WebTestUtils,
-        browser: Chrome,
-        sp_url: Callable[[ServiceProviderInstance], str],
-        secrets: TestSecrets,
-        netid: str,
-        login_transform: Callable[[str], str],
-        password_transform: Callable[[str], str],
-        expected_error_message: str):
-    """
-    :param login_transform: A function that accepts a string, and returns a string.
-    :param password_transform:  A function that accepts a string, and returns a string.
-    """
+def test_new_session_sign_on_domain_credential(browser, netid, utils, sp_url, secrets, sp_domain, login_transform):
+    # 7 New session, sign on with @domain credential format
+    fresh_browser = Chrome()
     login = login_transform(netid)
+    password = secrets.test_accounts.password
     sp = ServiceProviderInstance.diafine6
-    password = password_transform(secrets.test_accounts.password)
-
     with utils.using_test_sp(sp):
-        browser.get(f'{sp_url(sp)}/shibprod')
-        browser.send_inputs(login, password)
-        browser.click(Locators.submit_button)
-        browser.wait_for_tag('p', expected_error_message)
-
-
-# PWD 11
-def test_query_param_forwarding(utils: WebTestUtils, browser: Chrome, sp_url: Callable[[ServiceProviderInstance], str],
-                                secrets: TestSecrets, netid: str):
-    sp = ServiceProviderInstance.diafine6
+        fresh_browser.get(f'{sp_url(sp)}/shibprod')
+        fresh_browser.send_inputs(login, password)
+        fresh_browser.click(Locators.submit_button)
+        if 'google' not in login:
+            fresh_browser.wait_for_tag('h2', f'{sp_domain(sp)} sign-in success!')
+        else:
+            element = fresh_browser.wait_for_tag('p', 'Please sign in with your')
+            assert ('Please sign in with your UW NetID' in element.text)
+    fresh_browser.close()
+
+
+def test_case_insensitivity___query_parameters(browser, netid, utils, sp_url, secrets, sp_domain):
```

Comment:
Sorry for the wall of text, I am not sure of the intent of your naming convention here, so tried to tackle several options based on your goals. In particular, if you'd like to talk further (outside of the PR) about option #3 (the `@pytest.mark` option, I think it would really be a good thing to adopt here no matter the outcome of this PR. It'll make configuration and maintenance hella easier as time rolls on.)

We can chat outside of github if you want to chat about this; I didn't intend for it to grow so large, I just get excited about test strategies.

---

What is `___query_parameters` telling us?

I've never seen a triple underscore before, and I'm not sure we should invent a new  convention when clearer alternatives exist. All the following options can work in a variety of circumstances, and I'll leave it to you to decide how to proceed in this case. 

If your intent is to namespace tests for a given type, a class would be appropriate:

```
class TestQueryParameters:
    def test_case_insensitivity(self, params, and_other_fixtures):
         # . . . 
```

Or, if you would prefer absolute clarity over namespacing, you can write a whole sentence, which is something I've seen done often:

```
def test_query_parameters_are_case_insensitive(params, and_other_fixtures):
```

[this works well until you have very specific, long test names..., in which case, namespacing or refactoring is usually in order, which is fine for later.]

You can also always break related tests into separate files, (`test_query_parameters.py::test_case_insensitivity`), if you expect to be adding many more related ones.

And, saving the best (and not mutually exclusive) for last, you have an option to mark tests with a test case, so that you don't have to worry about trying to match the matrix naming conventions to the python naming conventions; this can then also be used to further break test cases out into modules that are easily runnable on their own. ("pytest run -k 'PWD10'), which gives us a lot of power!

```
@pytest.mark.PWD_10  # I realize this is not likely PWD10 related, that one is just in my head
def test_query_parameters_are_case_insensitive(params, other_fixtures):
   # . . .
```

This will _just work_*

*with a warning, until you add some config I can tell you about later, but it won't cause any problems until then!

Reply:
Thank for the thoughts. Before I get too far into this, my intention was to name a test function with the 2 tests that it covers. Test 8 is supposed to leave the existing browser window open and then proceed to Test 9, so I lumped them into one test to preserve the browser but maybe I made more work for myself. I could avoid the whole naming debacle if I could access `fresh_browser` in Test 9 that originated in Test 8. Looking into that :)

Reply:
In that case, you want a class, and a  fixture, which I think is part of the giant slack thread I left you the other day about different test approaches and may have more info:

```
@pytest.fixture(scope='class')
def class_browser() -> Chrome:
    browser = Chrome()
    yield browser
    browser.close()


class SomeNiceNameForTest8And9:
    @pytest.fixture(autouse=True)
    def initialize(class_browser):
        self.browser = class_browser

    def test_8():
         self.browser.do_work()
    
    def test_9():
        self.browser.is_still_where_it_was()

class SomeNiceNameForOtherCases:
    @pytest.fixture(autouse=True)
    def initialize(class_browser):
        self.browser  = class_browser  # This is fresh, and not the same one used for 8 and 9

```

Reply:
Exactly, I was just reading and implementing that exact slack message :)

Reply:
Yay!

Reply:
>class SomeNiceNameForTest8And9:
    @pytest.fixture(autouse=True)  # pytest class can't use __init__; this is the replacement
    def initialize(class_browser):
    
  def initialize(**self,** class_browser) # I think this needs self, or at least pycharm wants it.

Reply:
Yes, you're correct.